pkg/scuzz/sg_linux.go

```diff
@@ -212,22 +199,53 @@ func (s *SGDisk) newPacket(cmd Cmd, direction direction, timeout uint) *packet {
 	p.iovCount = 0 // is this ever non-zero?
 	p.dataLen = uint32(oldSchoolBlockLen)
 	p.nsect = 1
+	p.ataType = ataType
 
 	return p
 }
 
-func (s *SGDisk) UnlockRequest(password string, timeout uint, master bool) Request {
-	p := s.newPacket(securityUnlock, to, timeout)
+func (s *SGDisk) unlockPacket(password string, timeout uint, master bool) *packet {
+	p := s.newPacket(securityUnlock, to, lba48, timeout)
 	p.genCommandDataBlock()
-
 	if s.master {
 		p.block[1] = 1
 	}
 	copy(p.block[2:], []byte(password))
-	return &sgRequest{packet: p}
+	return p
 }
 
-func (s *SGDisk) Operate(r Request) error {
-	_, _, err := unix.Syscall(unix.SYS_IOCTL, uintptr(s.f.Fd()), r.Cmd(), r.Packet())
-	return err
+// Unlock performs unlock requests for Linux SCSI Generic Disks
+func (s *SGDisk) Unlock(password string, timeout uint, master bool) (string, error) {
+	p := s.unlockPacket(password, timeout, master)
+	if err := s.operate(p); err != nil {
+		return "", err
+	}
+	return "Unlock succeeded", nil
```

Comment:
here, I'd just return a nil error. no need to have two variables that indicate the inverse of each other